WordPress/gutenberg
# ComboboxControl: Normalize hyphen-like unicode characters to ASCII hyphens when matching search queries

## What?
When searching, treat hyphen-like characters as an ASCII hyphen (%2D).

## Why?
This is the implementation agreed upon in discussions on #41952. This improves the search but doesn't affect search result highlighting or cause any other accessibility regressions.

## How?
Following the [example outlined](https://github.com/WordPress/gutenberg/pull/41952#pullrequestreview-1037737092) by @ciampo , we have a list of hyphen-like characters that are replaced with an ASCII `-` when looking for search matches.

## Testing Instructions
Run
```
npm run test:unit ./packages/components/src/utils/test/strings.js
```
to verify string normalization is functioning properly.

## Screenshots or screencast 

![image](https://user-images.githubusercontent.com/917632/182648024-75798a4c-cb2b-423d-977c-f7a1e1bab980.png)

![image](https://user-images.githubusercontent.com/917632/182648041-9c0eb230-6d40-48d7-a6ab-cd1ddd8a6980.png)
